src/cargo/ops/cargo_rustc/commands.rs

```diff
@@ -0,0 +1,131 @@
+use std::collections::HashMap;
```

Comment:
In theory we could even provide an engine one day that links directly to `librustc` and sidesteps spawning processes altogether...